src/main/java/uk/gov/hmcts/ccd/domain/model/std/CaseDataContent.java

```diff
@@ -34,6 +34,9 @@
     @JsonProperty("case_reference")
     private String caseReference;
 
+    @JsonProperty("on_behalf_of_token")
+    private String onBehalfOfUserToken;
```

Comment:
Why whole userToken?? we only need invoker's userId right? 
it’s a bit odd to pass user auth token around systems rather than required userInfo elements.

Reply:
hi @smathangi initial request from architects was to pass the whole token. I also thought `userId` would be enough so we started with `userId`. The complication we found was that the IdamClient `getUserDetails` API wants the whole token in input to retrieve the user info, not the `userId`. 
Therefore we moved back to the initial suggestion of passing the whole token. 
I now also think that the benefit of passing the whole token is that the client can prove that he's authorised to use `on_behalf_of` because he has a valid token for that user
I'm not adamant on this though, I guess there are pros and cons for both approaches. What do you think?

Reply:
CCD api is not using that security token for authentication / authorisation purpose. So, to me what's the need to expose to CCD ?  Also one extra call to IDAM from ccd in the e2e user flow. Instead a caller (MCA) can actually pass all required bits (userid, first_name, lastName) in the payload.

I just thought of flagging this - either way is fine with me. (as architects already discussed )

Reply:
Just had a chat with Mario. For now we can keep the implementation as it is. 
It can be improved a bit later once architects give it a thought on new proposals ( 1. Caller (mca) pass valid data for all required fields . Or 2. Caller pass userId and CCD uses it's own system-user-token to retrieve userInfo for a given userId).